lib/utils/validateOptions.js

```diff
@@ -65,6 +65,17 @@ function validate(opts, ruleName, complain) {
 	const possible = opts.possible;
 	const actual = opts.actual;
 	const optional = opts.optional;
+	const whitelist = [
+		'reportDescriptionlessDisables',
+		'reportInvalidScopeDisables',
+		'reportNeedlessDisables',
+	];
```

Comment:
I think it's reasonable that `validateDisableSettings` uses internally `validateOptions` to re-use the logic of `validateOptions`:

https://github.com/stylelint/stylelint/blob/0bd6976fea7fa2a07c47672de424194716d31164/lib/validateDisableSettings.js#L50-L63

However, I don't think it's maintainable the `whitelist` array depends implicitly on the callees of `validateDisableSettings`; we would have to change two locations if we would add a new disable setting: 

https://github.com/stylelint/stylelint/blob/0bd6976fea7fa2a07c47672de424194716d31164/lib/invalidScopeDisables.js#L13

https://github.com/stylelint/stylelint/blob/0bd6976fea7fa2a07c47672de424194716d31164/lib/needlessDisables.js#L17

https://github.com/stylelint/stylelint/blob/0bd6976fea7fa2a07c47672de424194716d31164/lib/descriptionlessDisables.js#L16-L19

Is there a better way?

Reply:
> Is there a better way?

If there's a way to know the direct parent, yes there is.
In short, create a function `isRule`.

Reply:
Nice idea 👍🏼